src/connection/connection.rs

```diff
@@ -6,26 +6,32 @@ use tokio::net::TcpStream;
 
 use crate::codec::{decode, encode, Token};
 use crate::command::{Command, CommandError, CommandProcessor};
+use crate::server::Context;
 
 pub type ConnectionId = u64;
 
 pub struct Connection {
     id: ConnectionId,
     socket: TcpStream,
     addr: SocketAddr,
+    context: Context,
 }
 
 impl Connection {
-    pub fn new(id: ConnectionId, socket: TcpStream, addr: SocketAddr) -> Self {
-        Connection { id, socket, addr }
+    pub fn new(ctx: Context, id: ConnectionId, socket: TcpStream, addr: SocketAddr) -> Self {
+        Connection {
+            id,
+            socket,
+            addr,
+            context: ctx,
+        }
     }
 
     pub async fn handle(&mut self) -> std::io::Result<()> {
         log::info!("(id={}) accepting connection from {}", self.id, self.addr);
 
         let mut buffer = BytesMut::with_capacity(4 * 1024);
-        let cp = CommandProcessor {}; // TODO: this should probably be shared? or we send via a
-                                      // queue?
+        let cp = CommandProcessor::new(self.context.clone());
```

Comment:
Cloning `context` seems dangerous. It's currently innocuous, but future you could dump _anything_ in there. That could make this a very memory-intensive operation, or it could open the door for bugs if some sort of state information is included and gets out of sync.

Reply:
Good point. I've added a comment in 042e6f3 on Context explaining what we'll do to uphold Clone. I've chosen to implement Cloning since we're using it to provide dependencies and they seem like they'll all either be queues, Arc<Mutex<>>, or similar, which will require calling clone.

I look forward to discovering the way this was a horrible decision, though, later on :joy: